girder/models/collection.py

```diff
@@ -191,18 +192,57 @@ def fileList(self, doc, user=None, path='', includeMetadata=False,
                     folder, user, path, includeMetadata, subpath=True):
                 yield (filepath, file)
 
-    def subtreeCount(self, doc):
+    def subtreeCount(self, doc, includeItems=True, user=None, level=None):
         """
         Return the size of the folders within the collection.  The collection
         is counted as well.
 
         :param doc: The collection.
+        :param includeItems: Whether items should be included in the count.
+        :type includeItems: bool
+        :param user: If filtering by permission, the user to filter against.
+        :param level: If filtering by permission, the required permission level.
+        :type level: AccessLevel
         """
         count = 1
         folders = self.model('folder').find({
             'parentId': doc['_id'],
             'parentCollection': 'collection'
         })
-        count += sum(self.model('folder').subtreeCount(folder)
-                     for folder in folders)
+
+        if level is not None:
+            folders = self.filterResultsByPermission(
+                cursor=folders, user=user, level=level, limit=None)
+        count += sum(self.model('folder').subtreeCount(
+            folder, includeItems=includeItems, user=user, level=level)
+            for folder in folders)
         return count
+
+    def setAccessList(self, doc, access, save=False, recurse=False, user=None,
+                      progress=noProgress, setPublic=None):
+        """
+        Overrides AccessControlledModel.setAccessList to add a recursive
+        option. When true, this will set the access list on all subfolders to
+        which the given user has ADMIN access level. Any subfolders that the
+        given user does not have ADMIN access on will be skipped.
+        """
+        progress.update(increment=1, message='Updating ' + doc['name'])
+        if setPublic is not None:
+            self.setPublic(doc, setPublic, save=False)
+        doc = AccessControlledModel.setAccessList(self, doc, access, save=save)
+
+        if recurse and save:
```

Comment:
I'm slightly torn about this. 

If you pass in recurse=true and save=false, this block will not be executed, which makes sense since you are not returning the set of changed resources and you are not saving.  The only time I could see this not doing what you expect is if you want to load, modify, and save the same set of changed resources externally to this method, and also want them to have the updated accessList, but that seems a bit contorted.

After consideration, I think it makes sense as it is, but I wanted to list my concerns. :flags: